rust/src/core.rs

```diff
@@ -163,6 +166,83 @@ impl FilesystemScriptPrep {
     }
 }
 
+/// Some Fedora/RHEL kernels ship .hmac files with absolute paths inside,
+/// which breaks when we relocate them into ostree/.  This function
+/// changes them to be relative.
+///
+/// Fixes are pending to ensure the kernels are doing this from the start,
+/// but we can't rely on that landing for a while. In addition, there are
+/// many 3rd-party kernels that have replicated the behavior in the ARK
+/// kernel.spec.
+fn verify_kernel_hmac_impl(moddir: &Dir) -> Result<()> {
+    use std::os::unix::ffi::OsStrExt;
+
+    // FIXME: in 2023
+    // This method is intentionally a misnomer because it should eventually
+    // be changed to the "sanity check" (below). It currently patches absolute
+    // paths to give kernel package maintainers time to update their .spec files.
+
+    const SEPARATOR: &[u8] = b"  ";
+
+    let hmac_path = ".vmlinuz.hmac";
+
+    let hmac_contents = if let Some(f) = moddir.open_optional(hmac_path)? {
+        let mut f = BufReader::new(f);
+        let mut s = Vec::new();
```

Comment:
Any reason you changed this to be a `Vec` vs `String`?  I can't imagine that there are non-UTF-8 hmac files.

Reply:
`Path` uses `AsRef<OsStr>` because there's no real guarantee that OS paths are valid UTF-8. It's just some healthy pedantry that Rust std like to be explicit about.